Bourreau/app/models/bourreau_worker.rb

```diff
@@ -536,4 +546,120 @@ def check_for_tasks_stuck_in_ruby
+                                                    # the same time. Otherwise, the new task may be submitted
+                                                    # by several workers. See nice examples at:
+                                                    # http://www.codegnome.com/blog/2013/05/26/locking-files-with-ruby
+          file_content = file.read
+          submit_task_from_string(file_content,task)
+          file.close # This also releases the lock. 
+          File.delete(filename)
+        else
+          worker_log.info("Another worker is working on this file: ignoring it.")
+          file.close # In case the lock was not obtained, still close the file. 
+        end
+      rescue => ex
+        worker_log.info("Error while submitting new task: #{ex.message}.")
+        File.delete(filename) if File.exists?(filename)
+      end
+    end
+  end
+
+  # Creates and submits a task defined by a JSON object.
+  # Parameters:
+  # * json_string: A string containing a JSON object defining the task.
+  #                Example:
+  #                  {
+  #                    "tool-class": "CbrainTask::TestTool",
+  #                    "description": "A task running TestTool",
+  #                    "parameters": {
+  #                       "important_number": "123",
+  #                       "dummy_paramet4er": "432"
+  #                    }
+  #                  }
+  # * current_task: The current task from which the new task is submitted.
+  #                 Used to set the user, bourreau id and results data provider id
+  #                 for the new task.
+  # 
+  def submit_task_from_string json_string,current_task
+
+    # Parses JSON string and checks format
+    validate_json_string(json_string) # Raises an exception if string is not valid
+    new_task_hash = JSON.parse(json_string)
+    worker_log.info("Submitting task #{new_task_hash}")
+    
+    # Creates task
+    task_class_name = new_task_hash["tool-class"]
+    new_task        = CbrainTask.const_get(task_class_name).new # Raises an exception if tool class is not found
+
+    # Sets tool config among tool configs accessible by user of current task
+    accessible_tool_configs = ToolConfig.find_all_accessible_by_user(current_task.user)
+    tool_config_id          = new_task_hash["tool-config-id"]
+    if tool_config_id.blank?
+      # Sets tool config as the first one we find for class task_class_name
+      tool        = Tool.where(:cbrain_task_class => "#{task_class_name}").first
+      tool_config = accessible_tool_configs.where(:tool_id => tool.id).first
+    else
+      tool_config = accessible_tool_configs.find(tool_config_id)
+    end
+    raise "Cannot find accessible tool config for class #{task_class_name}" if tool_config.blank?
+    new_task.tool_config = tool_config
+
+    # Sets task parameters
+    new_task.params = Hash.new
+    new_task_hash["parameters"].each do |param|
```

Comment:
A value that is a JSON object will be parsed as a hash. However, I'm not sure it will correctly work because the resulting hash will look like {"foo"=>"bar", "plop"=>"coin"} while a CBRAIN task would rather use symbols instead of strings {:foo => "bar", :plop => "coin"}. At the first key-value level, strings are converted to symbols in method submit_task_from_string: this may have to be improved to cover multiple levels of key-value pairs.